tests/common/include/tests/common/tools/create_toy_geometry.hpp

```diff
@@ -9,432 +9,344 @@
-    using disc_container = detray::dvector<disc>;
-    using cylinder_container = detray::dvector<cylinder>;
-    using rectangle_container = detray::dvector<rectangle>;
-
-    /** Volumes */
-    volume_container volumes = {};
-    /** Surfaces, including portals */
-    surface_container surfaces = {};
-    /** Surface transforms */
-    transf_container transforms = {};
-    /** Disc masks for neg/pos volume boundaries (portals) */
-    disc_container discs = {};
-    /** Cylinder masks for inner/outer volume boundaries (portals) */
-    cylinder_container cylinders = {};
-    /** Rectangle masks for detector surfaces */
-    rectangle_container rectangles = {};
-    // mask index for surfaces
-    mask_index m_id = {};
-    /** source link */
-    const dindex inv_sf_finder = dindex_invalid;
-    ;
-    /** Leaving world */
-    const dindex leaving_world = dindex_invalid;
+    // detector type
+    using detector_t = detector<
+        darray, dtuple, dvector, djagged_vector,
+        static_transform_store<dvector>,
+        unified_index_geometry<dvector, darray, dtuple, dindex, dindex>,
+        serializer2, std::map<dindex, std::string> >;
+
+    // sub-geometry components type
+    using geometry = typename detector_t::geometry;
+    using volume = typename detector_t::volume;
+    using edge_links = typename detector_t::geometry::edge_links;
+    using surface = typename detector_t::surface;
+    using mask_container = typename detector_t::mask_container;
+    using transform_store = typename detector_t::transform_store;
+    using transform_container = typename detector_t::transform_container;
+    using disc = typename detector_t::geometry::disc;
+    using cylinder = typename detector_t::geometry::cylinder;
+    using rectangle = typename detector_t::geometry::rectangle;
+    using surface_container =
+        typename detector_t::geometry::surface_filling_container;
 
-    // parameters
-    scalar detector_half_z = 500.;
-    scalar beampipe_r = 27.;
-    scalar first_layer_outer_r = 38.;
-    scalar second_layer_inner_r = 64.;
-    scalar second_layer_outer_r = 80.;
-
-    struct toy_grid {};
-
-    /** Add a single barrel layer volume to an existing collection.
-     *
-     * @param min_r minimal radius of volume
-     * @param max_r maximal radius of volume
-     * @param half_z half length in z of volume
-     *
-     * @return a detray cylinder volume
+    /** Function that adds a disc portal.
      */
-    auto add_cylinder_volume = [&](const scalar min_r, const scalar max_r,
-                                   const scalar half_z) -> volume_type& {
-        // The volume bounds
-        detray::darray<scalar, 6> bounds = {min_r,  max_r, -half_z,
-                                            half_z, -M_PI, M_PI};
-        // Add the new volume to the collection
-        auto& vol = volumes.emplace_back(bounds);
-        vol.set_index(volumes.size() - 1);
-
-        return vol;
-    };
+    auto add_disc_portal =
```

Comment:
How much sense does it make to keep these as lambdas?

Reply:
I definitely prefer to have it as a member function in detector class. But let's leave it there for the moment until we sort out everything